golem-worker-service-base/src/service/api_definition_validator.rs

```diff
@@ -42,19 +47,30 @@ impl RouteValidationError {
 }
 
 #[derive(Clone)]
-pub struct ApiDefinitionValidatorDefault {
-    template_service: Arc<dyn TemplateService + Send + Sync>,
+pub struct ApiDefinitionValidatorDefault<Namespace, AuthCtx> {
+    template_service: Arc<dyn TemplateService<Namespace, AuthCtx> + Send + Sync>,
 }
 
-impl ApiDefinitionValidatorDefault {
-    pub fn new(template_service: Arc<dyn TemplateService + Send + Sync>) -> Self {
+impl<Namespace, AuthCtx> ApiDefinitionValidatorDefault<Namespace, AuthCtx> {
+    pub fn new(
+        template_service: Arc<dyn TemplateService<Namespace, AuthCtx> + Send + Sync>,
```

Comment:
As mentioned before, Validator could forget about any other interface such as template-service here. Being able to fetch the output with defined=-input than in-directions generally pay off better in my experience. Example: When writing a test Generating invalid template-metadata using `arbitrary` for a given `ApiDefinition` is easier with that approach. Otherwise you have to define a type, which takes ApiMetadata, and then write a template-service for it that always emits conflicting template metadata, and then having to figure out another type (a new type) for which it always succeeds etc and that seems to be an overkill.  Let me know what you think.